### ISSUE TITLE
Suboptimal tooltip positioning in facets

### ISSUE THREAD

**Issue by orausch** [JetBrains/lets-plot#1187]
I often have plots with many tooltips, and the current overflowing behavior makes these a bit tricky to use. Example:
```
from lets_plot import *

LetsPlot.setup_html()

d = {
    l: [i]*3 for i, l in enumerate(string.ascii_lowercase)
    
} | {'facet': [0, 1, 2]}
ggplot(d, aes(x='a', y='b')) + geom_point(tooltips=layer_tooltips(list(string.ascii_lowercase))) + ggsize(300, 1000) + facet_wrap('facet')
```
The lower tooltips have enough space to display:

<img width="754" alt="Screenshot 2024-09-16 at 8 24 28 AM" src="https://github.com/user-attachments/assets/a1e3ef29-932e-44ec-ae67-a98ea622398a">

The upper ones overflow even though there is space for them:
<img width="754" alt="Screenshot 2024-09-16 at 8 24 37 AM" src="https://github.com/user-attachments/assets/8852f705-cc03-47c9-9512-68cbb3bf9968">

**Comment by alshan** [comment]
Likely related: #1043

**Comment by IKupriyanov-HORIS** [comment]
We attempted to prevent overlap between tooltips (e.g., the axis tooltip 0.0 forced the general tooltip to align at the top).

In the fix, I relaxed the overlapping restriction. Now, the general tooltip can overlap other tooltips (e.g., the axis tooltip becomes hidden in such cases). This leaves room for improvements, such as using a semi-transparent general tooltip or assigning a higher z-order to axis tooltips.

Output with the fix:
![image](https://github.com/user-attachments/assets/3d9a7203-98c3-4f77-a4d4-d4e82d48e5a0)
